src/webots/vrml/WbProtoManager.cpp

```diff
@@ -105,85 +107,134 @@ void WbProtoManager::readModel(WbTokenizer *tokenizer, const QString &worldPath)
   model->ref();
 }
 
-WbProtoModel *WbProtoManager::findModel(const QString &modelName, const QString &worldPath, QStringList baseTypeList) {
+WbProtoModel *WbProtoManager::findModel(const QString &modelName, const QString &worldPath, const QString &parentFilePath,
+                                        QStringList baseTypeList) {
```

Comment:
We should avoid passing argument by value
```suggestion
                                        const QStringList &baseTypeList) {
```